src/solvers/refinement/string_refinement.cpp

```diff
@@ -1838,9 +1810,16 @@ class find_index_visitort: public const_expr_visitort
 static exprt find_index(
   const exprt &expr, const exprt &str, const symbol_exprt &qvar)
 {
-  find_index_visitort v(str, qvar);
-  expr.visit(v);
-  return v.index;
+  for(auto it=expr.depth_begin(); it!=expr.depth_end(); ++it)
+  {
+    if(expr.id()==ID_index)
+    {
+      const index_exprt &i=to_index_expr(expr);
```

Comment:
I'd rename `i` to `index`, as `i` is usually the name used by the incremented integer in a loop.